packages/grafana-schema/src/schema/mudball.gen.ts

```diff
@@ -473,12 +480,49 @@ export enum BarGaugeDisplayMode {
   Lcd = 'lcd',
 }
 
+/**
+ * Interface for table cell types that have no additional options.
+ */
+export interface TableCellNoOptions {
+  displayMode: TableCellDisplayMode;
+}
+
+/**
+ * Allows for the table cell gauge display type to set the gauge mode.
+ */
+export interface TableGaugeCellOptions {
+  displayMode: TableCellDisplayMode.Gauge;
+  gaugeDisplayMode: BarGaugeDisplayMode;
+}
+
+/**
+ * Allows for the background display mode to be set for the color background cell.
+ */
+export interface TableColorBackgroundCellOptions {
+  displayMode: TableCellDisplayMode.ColorBackground;
+  backgroundDisplayMode: TableCellBackgroundDisplayMode;
+}
```

Comment:
```suggestion
export interface TableAutoCellOptions {
  type: TableCellType.Auto;
}

/**
 * Allows for the table cell gauge display type to set the gauge mode.
 */
export interface TableBarGaugeCellOptions {
  type: TableCellType.BarGauge;
  mode: BarGaugeDisplayMode;
}

/**
 * Allows for the background display mode to be set for the color background cell.
 */
export interface TableColoredBackgroundCellOptions {
  type: TableCellDisplayMode.ColorBackground;
  mode: 'gradient' | 'solid';
}
```

Reply:
Maybe TableCellNoOptions is ok for to reduce number of types

Reply:
I changed it to `TableCellAutoOptions` but still referencing `TableCellDisplayMode` since the majority of table cells don't have extra configuration options at the moment.